### PR TITLE
Netscaler various fixes

### PR DESCRIPTION
**Pull request by giorgos-nikolopoulos**
##### SUMMARY

This PR introduces the following fixes and enhancements

* Add "state" suboption for netscaler_servicegroup servicemembers
* Fix servicemember modification algorithm in netscaler_servicegroup
* Fix monitorbindings modification algorithm in netscaler_servicegroup
* Add options for gracefully disabling netscaler_server
* Add default lb vserver option in netscaler_cs_vserver
* Add documentation for ssl_certkey option in netscaler_cs_vserver

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

* lib/ansible/modules/network/netscaler/netscaler_servicegroup.py
* lib/ansible/modules/network/netscaler/netscaler_server.py
* lib/ansible/modules/network/netscaler/netscaler_cs_vserver.py

##### ANSIBLE VERSION

```
ansible 2.5.0 (netscaler_various_fixes 787fc58c4b) last updated 2018/01/11 19:33:09 (GMT +300)
  config file = /home/georgen/.ansible.cfg
  configured module search path = [u'/home/georgen/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/georgen/repos/ansible_fork/lib/ansible
  executable location = /home/georgen/repos/ansible_fork/bin/ansible
  python version = 2.7.12 (default, Nov 20 2017, 18:23:56) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION

Unit tests were also updated for the relevant modules.

The following test files were modified.

* test/units/modules/network/netscaler/test_netscaler_cs_vserver.py
* test/units/modules/network/netscaler/test_netscaler_server.py
* test/units/modules/network/netscaler/test_netscaler_servicegroup.py

```

```

### CODE REVIEW

**Comment by giorgos-nikolopoulos** [comment]
@resmo, @gundalow , @privateip 

Would it be possible for any of you guys to review this PR in order to have it merged by the 2.5 deadline?

According to [this source](https://groups.google.com/forum/#!searchin/ansible-devel/release$20date%7Csort:date/ansible-devel/vGCQQY5472s/uRNgVO_rAwAJ) the deadline is February 8th for community modules.

I took the liberty of pinging you since you have reviewed Netscaler PRs in the past.
Thanks in advance for your time.

File: lib/ansible/modules/network/netscaler/netscaler_servicegroup.py [hunk 1]
```diff
@@ -271,6 +271,12 @@
                     - Server port number.
                     - Range C(1) - C(65535)
                     - "* in CLI is represented as 65535 in NITRO API"
+            state:
```
Comment (resmo) [inline lib/ansible/modules/network/netscaler/netscaler_servicegroup.py:274]: I would prefer another name for this param `state` for consistency with other modules, which uses state for `present` or `absent`. I would suggest to use `enabled=<bool>`. Any other opinions?

Comment (resmo) [reply lib/ansible/modules/network/netscaler/netscaler_servicegroup.py:274]: I just realized it is a suboption. LGTM

**Review by resmo — APPROVED** [review]
shipit

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merged into `devel` will be released in Ansible 2.5.0
Thanks!

**Comment by gundalow** [comment]
@giorgos-nikolopoulos Should you be listed as a maintainer for these directories, if so please raise a PR to update https://github.com/ansible/ansible/blob/devel/.github/BOTMETA.yml

**Comment by giorgos-nikolopoulos** [comment]
@resmo @gundalow Thanks for the prompt response.

I am mentioned along with @chiradeep as maintainers for the netscaler modules.

Does this mean we can call shipit on netscaler modules' PRs, even if they  are opened by us?

**Comment by gundalow** [comment]
@giorgos-nikolopoulos You can not add `shipit` on your own PR, well you can though `ansibullbot` will ignore it.

As long as you don't just give blanket `shipits` without actually reviewing the code we are all good :)

**Pull request merged**